### PR TITLE
fix: Maximum call stack size exceeded (#40)

### PR DESCRIPTION
**Pull request by milahu**
part of https://github.com/sveltejs/svelte/pull/6716

start fixing #40 and https://github.com/sveltejs/svelte/issues/4694

### CODE REVIEW

**Comment by Conduitry** [comment]
We don't want to be mutating `Array.prototype` to add our own new method there. This should either be implemented as a regular function that accepts the array as an argument, or it should use reassignments and `concat`.

**Comment by milahu** [comment]
> This should either be implemented as a regular function that accepts the array as an argument

yepp

```js
// original
a.push(...b, c)

// replaceMethod = "pushArray"
a.pushArray(b, [c])

// current version
// replaceMethod = "pushArrayNoMethod"
pushArray(a, b, [c])
```

> or it should use reassignments and `concat`.

thats usually slower

code-red has no eslint, so the patched code looks a bit ugly

**Review by benmccann — COMMENTED** [review]
could possibly tweak the naming a bit, but otherwise lgtm

File: src/utils/push_array.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export function push_array(thisArray, ...otherList) {
```
[the remark this answers is not in the document]
Comment (milahu) [reply src/utils/push_array.js:1]: otherList is a list of otherArray ... maybe call it otherArrayList?

example call:

```js
push_array(thisArray, otherArray1, otherArray2, otherArray3)
```

**Comment by milahu** [comment]
closing for #71 - thanks all : )

**Pull request closed without merging**